src/core/db/repository/task.py

```diff
@@ -25,11 +26,25 @@ async def get_all_user_tasks(self) -> list[Task]:
         """Получить список задач из категорий на которые подписан пользователь."""
         return await self._session.scalars(select(Task).options(joinedload(Task.category)))
 
-    async def get_tasks_limit_for_user(self, limit: int, offset: int) -> list[Task]:
+    async def get_tasks_limit_for_user(self, limit: int, offset: int, user_telegram_id: int) -> list[Task]:
         """Получить limit-выборку из списка всех задач пользователя."""
-        return await self._session.scalars(select(Task).options(joinedload(Task.category)).limit(limit).offset(offset))
+        user_id = await self._session.scalar((select(User.id).where(User.telegram_id == user_telegram_id)))
+        return await self._session.scalars(
+            select(Task)
+            .join(Category)
+            .options(joinedload(Task.category))
+            .where(and_(Category.users.any(id=user_id)), Task.is_archived == false())
+            .limit(limit)
+            .offset(offset)
+        )
 
-    async def get_user_tasks_count(self) -> int:
+    async def get_user_tasks_count(self, user_telegram_id: int) -> int:
         """Получить общее количество задач для пользователя."""
-        tasks = await self._session.scalars(select(Task).options(joinedload(Task.category)))
+        user_id = await self._session.scalar((select(User.id).where(User.telegram_id == user_telegram_id)))
+        tasks = await self._session.scalars(
+            select(Task)
+            .join(Category)
+            .options(joinedload(Task.category))
+            .where(and_(Category.users.any(id=user_id)), Task.is_archived == false())
```

Comment:
Давай вместо `and_` использовать дополнительный `where`
```suggestion
            .where(Category.users.any(id=user_id))
            .where(Task.is_archived == false())
```